npm.js

```
@@ -0,0 +1,53 @@
+import { build, emptyDir } from "https://deno.land/x/dnt/mod.ts";
+
+await emptyDir("./npm");
+
+const wasm = await Deno.readFile(
+  "../../Rust/wallet-core/pkg/dusk_wallet_core_bg.wasm"
+);
+const string = `let wasm = new Uint8Array([${wasm.join(
+  ","
+)}]);function initSync(){return wasm};export { initSync }`;
+const jsFile = await Deno.open(
+  "../../Rust/wallet-core/pkg/dusk_wallet_core.js"
+);
+
+await Deno.writeTextFile("./pkg/dusk_wallet_core.js", string);
+
+await build({
+  entryPoints: ["./src/mod.js"],
+  outDir: "./npm",
+  typeCheck: false,
+  declaration: false,
+  test: false,
+  mappings: {
+    "npm:fake-indexeddb": {
+      name: "fake-indexeddb",
+      version: "5.0.1",
+    },
+    "https://unpkg.com/dexie/dist/dexie.mjs": {
+      name: "dexie",
+      version: "3.2.4",
+    },
+  },
+  shims: [],
+  package: {
+    // package.json properties
+    name: "@dusk-network/dusk-wallet-js",
+    version: Deno.args[0],
```

The idea here is the following:

1. If no version is given from the command line, it uses the most recent (in terms of semver) tag that is present. If no valid tags exists, throws an error.

2. If a version is given from the command line, it checks if it exists in the tags present in the repo. If not, throws an error.

In both cases, if the operation succeed, the tool needs to temporary checkout the code at the valid tag given or found, build the library, and then check back the initial starting commit.

If the repo is "dirty" – there are changes that are not committed – the tool should throw an error.